src/ConnectedMode.UnitTests/Suppressions/ServerIssuesStoreTests.cs

```diff
@@ -30,6 +34,165 @@ public class ServerIssuesStoreTests
+
+        [TestMethod]
+        [DataRow(false)]
+        [DataRow(true)]
+        public void AddIssues_EmptyList_ResultContainsNewIssues(bool clearAllExistingIssues)
+        {
+            var issue1 = CreateIssue("1", false);
+            var issue2 = CreateIssue("2", false);
+
+            var testSubject = new ServerIssuesStore();
+
+            var result = testSubject.Get();
+            result.Count().Should().Be(0);
+
+            testSubject.AddIssues(new List<SonarQubeIssue>() { issue1, issue2 }, clearAllExistingIssues: clearAllExistingIssues);
+
+            result = testSubject.Get();
+            result.Count().Should().Be(2);
+            result.Should().Contain(issue1);
+            result.Should().Contain(issue2);
+        }
+
+        [TestMethod]
+        public void AddIssues_NonEmptyList_ClearExisting_OldIssuesCleared()
+        {
+            var issue1 = CreateIssue("1", false);
+            var issue2 = CreateIssue("2", false);
+
+            var testSubject = new ServerIssuesStore();
+
+            var result = testSubject.Get();
+            result.Count().Should().Be(0);
+
+            testSubject.AddIssues(new List<SonarQubeIssue>() { issue1 }, clearAllExistingIssues: false);
+
+            result = testSubject.Get();
+            result.Count().Should().Be(1);
+            result.Should().Contain(issue1);
+
+            testSubject.AddIssues(new List<SonarQubeIssue>() { issue2 }, clearAllExistingIssues: true);
+
+            result = testSubject.Get();
+            result.Count().Should().Be(1);
+            result.Should().Contain(issue2);
+        }
+
+        [TestMethod]
+        public void AddIssues_ListOfAddIssues_NonEmptyList_DontClearExisting_OldIssuesAreRetained()
+        {
+            var issue1 = CreateIssue("1", false);
+            var issue2 = CreateIssue("2", false);
+
+            var testSubject = new ServerIssuesStore();
+
+            var result = testSubject.Get();
+            result.Count().Should().Be(0);
+
+            testSubject.AddIssues(new List<SonarQubeIssue>() { issue1 }, clearAllExistingIssues: false);
+
+            result = testSubject.Get();
+            result.Count().Should().Be(1);
+            result.Should().Contain(issue1);
+
+            testSubject.AddIssues(new List<SonarQubeIssue>() { issue2 }, clearAllExistingIssues: false);
+
+            result = testSubject.Get();
+            result.Count().Should().Be(2);
+            result.Should().Contain(issue1);
+            result.Should().Contain(issue2);
+        }
+
+        [TestMethod]
+        public void AddIssues_AddNullIssues_EventIsNotInvoked()
+        {
+            var testSubject = new ServerIssuesStore();
+
+            var eventMock = new Mock<EventHandler>();
+            testSubject.ServerIssuesChanged += eventMock.Object;
+
+            testSubject.AddIssues(null, false);
+
+            eventMock.VerifyNoOtherCalls();
+        }
+
+        [TestMethod]
+        public void AddIssues_EventIsInvoked()
```

Comment:
Being really picky here - logically this test only checks that no event is raised if `clearAllExistingIssues` is `false`. It would be easy to make this a data-driven test that covered both cases.